File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentSubtitleColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentSubtitleRow } from "./ContentSubtitleRow";

export interface ContentSubtitleColumns {
    Id: Column<ContentSubtitleRow>;
    SubtitleFile: Column<ContentSubtitleRow>;
    ContentTitle: Column<ContentSubtitleRow>;
    LanguageName: Column<ContentSubtitleRow>;
    Username: Column<ContentSubtitleRow>;
}

export class ContentSubtitleColumns extends ColumnsBase<ContentSubtitleRow> {
    static readonly columnsKey = 'Content.ContentSubtitle';
    static readonly Fields = fieldsProxy<ContentSubtitleColumns>();
}

File: GXpert/GXpert.Web/Modules/Content/ContentSubtitle/ContentSubtitleColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentSubtitle")]
[BasedOnRow(typeof(ContentSubtitleRow), CheckNames = true)]
public class ContentSubtitleColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string SubtitleFile { get; set; }
    public string ContentTitle { get; set; }
    public string LanguageName { get; set; }
    public string Username { get; set; }
   
}